nemo/collections/nlp/modules/common/megatron/clip_grads.py

```diff
@@ -190,6 +190,8 @@ def clip_grad_norm_distributed_optimizer(optimizer, max_norm, norm_type=2):
     # Compute grad norm
     # Note: Compute norm of local grads and sum over all procs
     grad_norm_sq = optimizer._local_grad_norm(parameters=params_for_norm, norm_type=norm_type)
+    if optimizer.redundant_size > 1:
+        grad_norm_sq /= optimizer.redundant_size
```

Comment:
Do all optimizers have this attribute? Can we do a safer check here?

Reply:
This grad clipping function guarantees that it's just dealing with the distributed optimizer:
https://github.com/NVIDIA/NeMo/blob/e6260265693dce0ac17b5ea9a26e77f910c7d2e9/nemo/collections/nlp/modules/common/megatron/clip_grads.py#L177

Reply:
I've also been thinking about moving this logic to the distributed optimizer wrapper at:
https://github.com/NVIDIA/NeMo/blob/fd6bb1d3fe6a74619cb41f9289b62af6252141c4/nemo/core/optim/distributed_adam.py#L21
I've been working on supporting BF16 grad reductions, so perhaps this work can be folded into that PR.